v4/feature/portfolio/src/main/java/exchange/dydx/trading/feature/portfolio/components/pendingpositions/DydxPortfolioPendingPositionItemView.kt

```diff
@@ -0,0 +1,177 @@
+        )
+    }
+}
+
+object DydxPortfolioPendingPositionItemView {
+    data class ViewState(
+        val id: String,
+        val localizer: LocalizerProtocol,
+        val logoUrl: String? = null,
+        val marketName: String? = null,
+        val margin: String? = null,
+        val viewOrderAction: () -> Unit = {},
+        val cancelOrderAction: () -> Unit = {},
+    ) {
+        companion object {
+            val preview = ViewState(
+                localizer = MockLocalizer(),
+                id = "1",
+                logoUrl = "https://media.dydx.exchange/currencies/eth.png",
+                marketName = "Ethereum",
+                margin = "1000.00",
+            )
+        }
+    }
+
+    @Composable
+    fun Content(modifier: Modifier, state: ViewState?) {
+        if (state == null) {
+            return
+        }
+
+        val shape = RoundedCornerShape(10.dp)
+        Column(
+            modifier = modifier
+                .fillMaxWidth()
+                .clip(shape)
+                .themeColor(ThemeColor.SemanticColor.layer_3),
+        ) {
+            Column(
+                modifier = Modifier
+                    .padding(
+                        horizontal = ThemeShapes.HorizontalPadding,
+                        vertical = 12.dp,
+                    ),
+                verticalArrangement = Arrangement.spacedBy(10.dp),
+            ) {
+                Row(
+                    verticalAlignment = Alignment.CenterVertically,
+                    horizontalArrangement = Arrangement.spacedBy(8.dp),
+                ) {
+                    PlatformRoundImage(
+                        icon = state.logoUrl,
+                        size = 20.dp,
+                    )
+
+                    Text(
+                        text = state.marketName ?: "",
+                        style = TextStyle.dydxDefault
+                            .themeFont(fontSize = ThemeFont.FontSize.base)
+                            .themeColor(ThemeColor.SemanticColor.text_secondary),
+                    )
+                }
+
+                Row(
+                    verticalAlignment = Alignment.CenterVertically,
+                    horizontalArrangement = Arrangement.spacedBy(ThemeShapes.HorizontalPadding),
+                ) {
+                    Text(
+                        text = state.localizer.localize("APP.GENERAL.MARGIN"),
+                        style = TextStyle.dydxDefault
+                            .themeFont(fontSize = ThemeFont.FontSize.small)
+                            .themeColor(ThemeColor.SemanticColor.text_tertiary),
+                    )
+
+                    Spacer(modifier = Modifier.weight(1f))
+
+                    Text(
+                        text = state.margin ?: "",
+                        style = TextStyle.dydxDefault
+                            .themeFont(fontSize = ThemeFont.FontSize.small)
+                            .themeColor(ThemeColor.SemanticColor.text_primary),
```

Comment:
non-blocking: i think this is secondary in the design

Reply:
OK.. will update in the next PR